### DISCUSSION TITLE
Database error saving new user

### DISCUSSION THREAD

**Discussion by fadethegap** [supabase/supabase#5289] [Questions]
Is it possible I corrupted my database?

Out of the blue my signUp() and signIn() functions stopped working. I started to receive the error: Database error saving new user.

In the Auth Users table when I click the Invite button and enter my email:

![Screen Shot 2022-02-02 at 3 04 00 PM](https://user-images.githubusercontent.com/47059192/152228718-6aa4373f-4e4d-4d45-9b37-8ba7711f1795.png)

I get the same error:

![Screen Shot 2022-02-02 at 2 57 06 PM](https://user-images.githubusercontent.com/47059192/152228754-3365a4ae-3911-4a62-b8d7-40df3c6bc965.png)

Any ideas on how to remedy this?

**Comment by fadethegap** [comment]
I ended up blowing away the project and rebuilding my tables, policies, functions, triggers, and function hooks. My code worked fine again. 

Then the same thing happened. I'm getting the same errors as above.

What do you think is happening?

**Reply by samsrose** [reply]
Yep. Sam exact thing. Supabase was awesome up until this point.

**Reply by reyhaan** [reply]
It would be a good idea to check if something's wrong with your sql, I was going crazy over the same error, I checked the logs and found there was an error with one of my triggers to write my own users table on creating a new user in auth.users table.

**Reply by fadethegap** [reply]
Thank you @reyhaan. Yes, that was the issue.

**Reply by yndotdev** [reply]
I didn't even know the Postgres logs existed. Thank you, @reyhaan!

**Reply by GaryAustin1** [reply]
@sammartfrank Using the dashboard, database tab, Postgres logs.  
You can also send your own messages to them with `raise log 'my message = %', variable;` in your functions.

**Comment by fadethegap** [comment]
As it turns out, the error was caused by an incorrectly configured Function Hook. My api call address was incorrect.

**Comment by Proziam** [comment]
For the future users who are googling for an answer:

`Database error saving new user` can occur if you have a supabase database function that involves a field that no longer exists (or has been renamed). For example, the quickstart guide for SvelteKit will cause this if you choose to delete the `avatar_url` field in your `profiles` table.

To quickly check this go to your supabase dashboard > Sidebar > Database > Functions

**Reply by tiffenyk** [reply]
This! Thanks a lot. To fix this, I edited the function "handle_new_user" with the correct name table.

**Reply by ghost** [reply]
This issue made me so angry. Thank you a lot for giving me a solution to resolve it.❤️

**Reply by Maxiviper117** [reply]
Jeez, wish the error message was more descriptive

**Reply by dlwocks31** [reply]
Thanks a lot, this saved by day.

I ended up replacing two database functions to solve this issue. At first I just tried to delete these two function, but it said something like I cannot because something depends on those.

- `authorize`:
```plpgsql

  -- declare
  --   bind_permissions int;
  -- begin
  --   select
  --     count(*)
  --   from public.role_permissions
  --   inner join public.user_roles on role_permissions.role = user_roles.role
  --   where
  --     role_permissions.permission = authorize.requested_permission and
  --     user_roles.user_id = authorize.user_id
  --   into bind_permissions;

  --   return bind_permissions > 0;
  -- end;

declare
  bind_permissions int;
begin
  -- No-op replacement: set bind_permissions to a fixed value
  bind_permissions := 0;

  return bind_permissions > 0;
end;
```

- `handle_new_user`:
```plpgsql

-- begin
--   insert into public.profiles (id, full_name, avatar_url)
--   values (new.id, new.raw_user_meta_data->>'full_name', new.raw_user_meta_data->>'avatar_url');
--   return new;
-- end;
begin
  -- No-op: removed the insert statement to make it a no operation

  return new;
end;
```

**Comment by silentworks** [comment]
I have outlined how to best check what this error is in the troubleshooting section of this discussion board. https://github.com/orgs/supabase/discussions/13043
